### PR TITLE
Turn animation + Move camera in start position for MOBs

### PR DESCRIPTION
**Pull request by jangernert**

### CODE REVIEW

**Comment by degenerated1123** [comment]
Sounds awesome! I'll quickly test those. Would you be okay with me adding you as a team-member so you can work on the main-repo?

**Comment by jangernert** [comment]
Sure, that would be nice. But I would mostly do smaller fixes like I have been doing so far. I don't think I'll have the time to work on bigger features myself.

**Comment by Kaffeine** [comment]
LGTM, but the second commit fixes the first one, so they should be squashed before push.

```git rebase -i HEAD~3```, then replace ```pick b6c2547``` by ```f b6c2547``` and ```git push -f``` to update the PR.

**Comment by jangernert** [comment]
Just out of curiosity: why is having 2 commits instead of one a problem?

**Comment by Aquaz** [comment]
If someone checks out the first commit for some reason and tries to fork from this point on they will have a bugged branch. Other than that in reality someone could decide to cut off a release build at the first commit which would result in a buggy release, this wouldn't happen in our case of course for obvious reasons but it's a good practice I suppose.

**Comment by Kaffeine** [comment]
1. Clear commits make it easier for others to study your code.
1.1 In a number of big projects with busy reviewers it is a "must" thing for PR to have no commits, which remove lines, added in a previous commit in the same PR. Exceptions — if you're add a stub or skeleton functions, which are implemented later commits. After all, it is a courteous to save reviewers time.
1.2 Probably one of the reasons to work on this (and most of hobbies) project is to learn something new. Clear history of sources attract other developers to study on your work.
1.3 Sometimes it is hard for a new developer to dive into a big tree of sources, but it is much easier to analyze atomic commits.
2. Proper intermediate commits are important for git bisect.

>If someone checks out the first commit for some reason and tries to fork from this point on they will have a bugged branch. Other than that in reality someone could decide to cut off a release build at the first commit which would result in a buggy release, this wouldn't happen in our case of course for obvious reasons but it's a good practice I suppose.

I don't think that someone will try to make a build at the first commit, but there is a significant chance that ```git bisect``` will do it and fail there.

I did a quick search and found this interesting link: http://dieter.plaetinck.be/post/why-rewriting-git-history-and-why-commits-imperative-present-tense/

I'm not fully agree with the author, but some of his arguments makes sense.

This is my personal opinion without any kind of offence. Everyone is welcome to agree or disagree and list reasons here. :)

**Comment by wisien92** [comment]
I doubt that people will fork feature-branch, since while it's not merge probably the feature is either not completed or not tested.

But as @Kaffeine said it's a matter of preference. For me, it's easier to look up and find particular change on a branch  that's commited often, with short descriptive messages. I don't like when there is ton of feature inside single commit and you have to browse thru files to find changes. But then again sometimes too often is also unreadable.

**Comment by degenerated1123** [comment]
Regarding the actual new code provided here:

I tested the new features and found them to be working fine. The only issue I found with the turning-animations were that the playermodel jumps up/down a few units when turning into the right direction. 
That's a limitation the engine currently has and @astillich is working on what makes a fix possible.

Reason for that (you see it when talking to people as well) is that I just assume some flags for all animation-files. Some need to be able to modify the root-position of the object, some don't. But since I forced the flag, all of them do.

So, if you're done discussing git, this can be merged I guess.

**Comment by jangernert** [comment]
@degenerated1123 I already noticed the same issue during other animations (can also be seen during daily routines) and saw a comment in the code somewhere.
But sometimes I get a segfault at the start because of "asMobVob". If I didn't know better I would assume something has to be initialized to use the function and I created some kind of race-condition. My C++ is a bit rusty. But isn't asMobVob() a stateless function? m_World is used even before, so I would assume everything is initialized already.

This is the line that is causing problems

```
Components::EntityComponent entity = alloc.getElement<Components::EntityComponent>(e);
```

```
#0  Vob::asVob (world=..., e=...) at /home/jeanluc/Projects/REGoth/src/components/Vob.cpp:47
#1  0x00000000004a4b71 in VobTypes::asMobVob (world=..., e=...) at /home/jeanluc/Projects/REGoth/src/components/VobClasses.cpp:147
#2  0x00000000004f5df5 in Logic::CameraController::onUpdateExplicit (this=0x11a3e30, deltaTime=0) at /home/jeanluc/Projects/REGoth/src/logic/CameraController.cpp:285
#3  0x00000000004acabd in Engine::GameEngine::onFrameUpdate (this=0xe19e30, dt=0, width=<optimized out>, height=<optimized out>)
    at /home/jeanluc/Projects/REGoth/src/engine/GameEngine.cpp:88
#4  0x000000000049ff86 in ExampleCubes::update (this=0x7fffffffdc80) at /home/jeanluc/Projects/REGoth/src/target/REGoth.cpp:851
#5  0x00000000004af28a in Engine::PlatformGLFW::run (this=this@entry=0x7fffffffdc80, argc=argc@entry=1, argv=argv@entry=0x7fffffffddd8)
    at /home/jeanluc/Projects/REGoth/src/engine/PlatformGLFW.cpp:270
#6  0x000000000049368a in main (argc=1, argv=0x7fffffffddd8) at /home/jeanluc/Projects/REGoth/src/target/REGoth.cpp:947

```

**Comment by degenerated1123** [comment]
You'd need to add a check for the mob-handle to be valid I believe. Should probably add a check for that in Vob::asVob.

**Comment by jangernert** [comment]
@degenerated1123 like this?
Also should this go into the same commit as well, since it's technically not directly related to the camera-feature.

File: src/components/Vob.cpp [hunk 1]
```diff
@@ -43,32 +43,32 @@ Handle::EntityHandle Vob::constructVob(World::WorldInstance& world)
...
     info.entity = e;
     info.world = &world;
 
+    if(!e.isValid())
+    {
+        return info;
+    }
+
+    Components::ComponentAllocator& alloc = world.getComponentAllocator();
+    Components::EntityComponent entity = alloc.getElement<Components::EntityComponent>(e);
```
Comment (degenerated1123) [inline src/components/Vob.cpp]: I forgot the & here. Mind adding it?

File: src/components/Vob.cpp [hunk 2]
```diff
@@ -43,32 +43,32 @@ Handle::EntityHandle Vob::constructVob(World::WorldInstance& world)
...
-
+    info.logic = nullptr;
+    info.visual = nullptr;
+    info.object = nullptr;
+    info.position = nullptr;
     info.entity = e;
     info.world = &world;
 
+    if(!e.isValid())
+    {
```
Comment (degenerated1123) [inline src/components/Vob.cpp]: No braces needed on single-line ifs like that :)

**Comment by jangernert** [comment]
Done

**Comment by degenerated1123** [comment]
Awesome! Does hat fix the crashes you were experiencing on start sometimes? If so, I'll merge it.

**Comment by jangernert** [comment]
Since I don't want to calculate the significance of my possible answer I'll just say yes :P 
Haven't experienced any crashes in the last ~15 tries.

**Pull request merged**